### PR TITLE
Update 2019 data

### LINKED ISSUE #47: Update power-profiler with 2019 data
The 2019 eGrid data is available.  Note: Puerto Rico has been added so we will need to make sure everything works with this addition.

### LINKED ISSUE #51: Fix the sub region map to display Puerto Rico properly

### PR DESCRIPTION
**Pull request by maheese**
This pull request closes #47, closes #51.  The changes here are in three areas: 1) updating the data to 2019, 2) fixing the links to point eGRID 2019, 3) fixing issues related to the addition of Puerto Rico (mainly maps).

1) I updated the prep scripts to work with the eGRID 2019 data.  The addition of PR caused problems in a couple of scripts.  Also there was a change in the name of the subregion property in the eGRID mapping files that was addressed.  I did not check in any of the data files.  If you run the scripts you can download the data from the eGRID website and rerun.  There is an issue with the mapping files in that PRMS appears twice which causes double entries in the GeoJSON file that's generated.  You will need to delete the first entry manually after the file is generated.  Travis is fixing this in the posted mapping files.  As a convenience I added a bash shell script that calls the prep scripts in order, so you don't have to do this manually.  This is a stopgap until we develop a better way of prepping the data.

2) I think I updated all of the links that point to the eGRID data and also the links that point to the eGRID subregion map located on the eGRID site.

3) With the addition of Puerto Rico the map projection had to be updated to one that included Puerto Rico.  I also adjusted the centroid for the label position.

### CODE REVIEW

File: app/src/components/modals/SubregionsModal.vue [hunk 1]
```diff
@@ -12,7 +12,7 @@
 
             <div class="modal-body">
                 <slot name="body">
-                    <p>The 26 eGRID subregions in the US are defined by EPA using data from the Energy Information Administration (EIA) and the North American Electric Reliability Corporation (NERC). The subregions are defined to limit the amount of imports and exports across regions in order to best represent the electricity used in each of the subregions. More information can be found in section 3.4.2 of the <a aria-label="Opens new window" href="/energy/egrid-technical-support-document" target="_blank">eGRID Technical Support Document</a>.</p>
+                    <p>The 27 eGRID subregions in the US are defined by EPA using data from the Energy Information Administration (EIA) and the North American Electric Reliability Corporation (NERC). The subregions are defined to limit the amount of imports and exports across regions in order to best represent the electricity used in each of the subregions. More information can be found in section 3.4.2 of the <a aria-label="Opens new window" href="/energy/egrid-technical-support-document" target="_blank">eGRID Technical Support Document</a>.</p>
```
Comment (j-tafoya) [inline app/src/components/modals/SubregionsModal.vue:15]: Nice catch!

File: prep/add_fuel_breakdown.py [hunk 2]
```diff
@@ -42,6 +42,6 @@
 df["Percent Hydro net generation"] = sn["SRHYPR"]
 # Add national data
 row = ["U.S.", "National", n["USTRPR"].values[0], n["USTNPR"].values[0], (n["USTNPR"] - n["USNCPR"]).round(3).values[0], n["USTHPR"].values[0], n["USNCPR"].values[0], n["USHYPR"].values[0]]
-df.loc[26] = row
+df.loc[len(df.index)] = row
```
Comment (j-tafoya) [inline prep/add_fuel_breakdown.py:45]: 🙌

File: prep/add_states.js [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
+var egrid_file = fs.readFileSync('data/shape/eGRID2019_subregions.json');
 var states_file = fs.readFileSync('data/shape/states.json');
 var egrid_j = JSON.parse(egrid_file);
 var states_j = JSON.parse(states_file);
 // Replace "Subregions" with new key: "name."
 for(var i = 0; i < egrid_j.features.length; i += 1){
-    egrid_j.features[i].properties["name"] = egrid_j.features[i].properties.Subregions;
-    delete egrid_j.features[i].properties["Subregions"];
+    egrid_j.features[i].properties["name"] = egrid_j.features[i].properties.ZipSubregi;
+    delete egrid_j.features[i].properties["ZipSubregi"];
```
Comment (j-tafoya) [inline prep/add_states.js:10]: So we'll need to change this once Travis gets this fixed right?

**Review by j-tafoya — COMMENTED** [review]
It looks like the SRVC region encompassing Virginia and the Carolinas is missing from the subregion.json file, so it excludes it from the maps and dropdown.
image

Maybe in the add_egrid_data.py? I'm still looking for where it might have been lost

File: app/src/components/modals/SubregionsModal.vue (hunk 1 above)
Comment (maheese) [reply app/src/components/modals/SubregionsModal.vue:15]: Thanks for the thorough testing!   I had not committed the most recent file.  I just pushed it to the branch.

**Review by j-tafoya — APPROVED** [review]
Looks good with updated subregion.json file. Good to merge

File: prep/add_states.js (hunk 3 above)
Comment (maheese) [reply prep/add_states.js:10]: Yep, I did point this out to Travis.  I don't think it will be fixed in the 2019 files, maybe next release?

**Pull request merged**